core/src/main/java/org/apache/iceberg/util/StructLikeMap.java

```diff
@@ -171,4 +172,10 @@ public R setValue(R value) {
       throw new UnsupportedOperationException("Does not support setValue.");
     }
   }
+
+  public <U> StructLikeMap<U> transformValues(Function<T, U> func) {
```

Comment:
Your idea of making this a static method also makes sense to me. It would be more in line with Guava and other similar libs. If you decide to make it static, keep in mind the ordering of methods. Static methods should be right after the constructor.

```
public static <T, U> StructLikeMap<U> transformValues(StructLikeMap<T> map, Function<T, U> func) {
  ...
}
```